quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use hyper::StatusCode;
+use quickwit_jaeger::JaegerService;
+use quickwit_opentelemetry::otlp::TraceId;
+use quickwit_proto::jaeger::storage::v1::span_reader_plugin_server::SpanReaderPlugin;
+use quickwit_proto::jaeger::storage::v1::{
+    FindTraceIDsRequest, GetOperationsRequest, GetServicesRequest, GetTraceRequest,
+    SpansResponseChunk, TraceQueryParameters,
+};
+use quickwit_proto::tonic;
+use quickwit_proto::tonic::Request;
+use tokio_stream::StreamExt;
+use warp::{Filter, Rejection};
+
+use crate::jaeger_api::model::{
+    JaegerError, JaegerResponseBody, JaegerSearchBody, TracesSearchQueryParams,
+};
+use crate::json_api_response::JsonApiResponse;
+use crate::{with_arg, BodyFormat};
+
+/// Setup Jaeger API handlers
+///
+/// This is where all Jaeger handlers
+/// should be registered.
+pub(crate) fn jaeger_api_handlers(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    jaeger_services_handler(jaeger_service_opt.clone())
```

Comment:
should I pass `Option<JaegerService>`  further down or it's better to unwrap Option here and respond with something like `internal_jaeger_error()` for the `None` case?